code/modules/research/stock_parts.dm

```diff
@@ -14,10 +14,41 @@
 	display_contents_with_number = 1
 	max_w_class = 3
 	max_combined_w_class = 100
+	var/works_from_distance = 0
+	var/pshoom_or_beepboopblorpzingshadashwoosh = 'sound/items/rped.ogg'
+	var/alt_sound = null
+
+/obj/item/weapon/storage/part_replacer/afterattack(obj/machinery/T as obj, mob/living/carbon/human/user as mob, flag, params)
+	if(flag)
+		return
+	else
+		if(works_from_distance)
+			if(istype(T))
+				if(T.component_parts)
+					T.exchange_parts(user, src)
+					user.Beam(T,icon_state="rped_upgrade",icon='icons/effects/effects.dmi',time=5)
+	return
```

Comment:
WHy are you doing this in the usual rpeds afterattack when you can do it in the bluespace one and remove the "works from distance" var?